server/src-lib/Hasura/Tracing.hs

```diff
@@ -208,16 +206,15 @@ extractHttpContext hdrs = do
     <*> pure freshSpanId
     <*> pure (HTTP.parseHeaderMaybe =<< lookup "X-Hasura-SpanId" hdrs)
 
-traceHttpRequest
-  :: MonadTrace m
-  => Text
-  -- ^ human-readable name for this block of code
-  -> m (SuspendedRequest m a)
-  -- ^ an action which yields the request about to be executed and suspends
-  -- before actually executing it
+-- | Perform HTTP request which supports Trace headers
+tracedHttpRequest 
+  :: MonadTrace m 
+  =>  HTTP.Request
+  -- ^ http request that needs to be made
+  -> (HTTP.Request -> m a)
+  -- ^ a function that takes the traced request and executes it
   -> m a
-traceHttpRequest name f = trace name do
-  SuspendedRequest req next <- f
+tracedHttpRequest req f = trace (T.decodeUtf8 (HTTP.path req)) do
```

Comment:
We should use `bsToTxt'` here, because `decodeUtf8` throws exceptions in the event of a decoding error. It's very unlikely, but we should guard against it.

Reply:
Fixed it :)